elastic/kibana
# [EPM] Improve object handling in index template mappings

## Summary

Implements https://github.com/elastic/kibana/issues/60492

Based on https://github.com/elastic/kibana/pull/63112, only merge after that one.

This adds handling for the `dynamic` and `enabled` attributes for fields of type `object` to the mapping used in index templates.

NOTE: the corresponding code in `libbeats` (in `beats/libbeat/template/processor.go`) does way more, but it reads fields that are not present in any beats modules, and it also does not unit test the processing of these additional attributes. Hence, this PR only implements what is actually unit-tested in  `beats/libbeat/template/processor_test.go`.

## How to test this

* Inspect the unit tests if they make sense.
* Package installation should still work, index templates should be created.
